spec/lib/payment_calculator/turnip_steps/ecf/output_payments_steps.rb

```diff
@@ -33,11 +33,13 @@ class DummyClass
       expect(@result.length).to eq(@retention_table.length)
       @retention_table.length.times.each do |x|
         key = @retention_table[x][:payment_type]
-        actual_values = @result[x][:output_payment][key]
-        expectation = @retention_table[x]
-        expect_with_context(actual_values[:retained_participants], expectation[:retained_participants], "#{expectation[:payment_type]} retention numbers passthrough")
-        expect_with_context(actual_values[:per_participant], expectation[:expected_per_participant_output_payment], "#{expectation[:payment_type]} per participant payment")
-        expect_with_context(actual_values[:subtotal], expectation[:expected_output_payment_subtotal], "#{expectation[:payment_type]} output payment")
+        @result[x][:output_payments].each_with_index do |output_payment, _index|
+          actual_values = output_payment[key]
+          expectation = @retention_table[x]
+          expect_with_context(actual_values[:retained_participants], expectation[:retained_participants], "#{expectation[:payment_type]} retention numbers passthrough")
+          expect_with_context(actual_values[:per_participant], expectation[:expected_per_participant_output_payment], "#{expectation[:payment_type]} per participant payment")
+          expect_with_context(actual_values[:subtotal], expectation[:expected_output_payment_subtotal], "#{expectation[:payment_type]} output payment")
+        end
```

Comment:
I know it was already here, but can we have a better name of `x`?

Also, `index` is not used, I think, so let's not use `each_with_index`.

Reply:
Refactored be6fa50